src/datajudge/constraints/stats.py

```diff
@@ -14,44 +16,73 @@ def __init__(
         self.significance_level = significance_level
         super().__init__(ref, ref2=ref2)
 
+    def retrieve(
+        self, engine: sa.engine.Engine, ref: DataReference
+    ) -> Tuple[Any, OptionalSelections]:
+        sel = ref.get_selection(engine)  # table selection incl. WHERE condition
+        col = ref.get_column(engine)  # column name
+        return sel, col
+
     @staticmethod
-    def calculate_2sample_ks_test(data: Collection, data2: Collection) -> float:
+    def approximate_p_value(d: float, m: int, n: int) -> Optional[float]:
         """
-        For two given lists of values calculates the Kolmogorov-Smirnov test.
-        Read more here: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html
+        Calculates the approximate p-value according to
+        'A procedure to find exact critical values of Kolmogorov-Smirnov Test', Silvia Fachinetti, 2009
         """
-        try:
-            from scipy.stats import ks_2samp
-        except ModuleNotFoundError:
-            raise ModuleNotFoundError(
-                "Calculating the Kolmogorov-Smirnov test relies on scipy."
-                "Therefore, please install scipy before using this test."
+
+        n = m + n // 2
+        if n < 35:
+            warnings.warn(
+                "Approximating the p-value is not accurate enough for sample size < 35"
             )
+            return None
 
-        # Currently, the calculation will be performed locally through scipy
-        # In future versions, an implementation where either the database engine
-        # (1) calculates the CDF
-        # or even (2) calculates the KS test
-        # can be expected
-        statistic, p_value = ks_2samp(data, data2)
+        d_alpha = d * math.sqrt(n)
+        approx_p = 2 * math.exp(-(d_alpha**2))
 
-        return p_value
+        # clamp value to [0, 1]
+        return 1.0 if approx_p > 1.0 else 0.0 if approx_p < 0.0 else approx_p
 
-    def retrieve(
-        self, engine: sa.engine.Engine, ref: DataReference
-    ) -> Tuple[Any, OptionalSelections]:
-        return db_access.get_column(engine, ref)
+    @staticmethod
+    def check_acceptance(d: float, n: int, m: int, accepted_level):
```

Comment:
Unless d, n, m are well-known shortcuts in statistics I'd go with proper variable names here and have a note what's the equivalent on that Wikipedia page, eg.

```
def ... sample1_size: int, sample2_size: int ...
"""
See [wikipedia url] for reference with n = sample1_size, m = sample2_size ...
"""

Reply:
Also a docstring for this function would be great

Reply:
I do agree with both comments, however, since `check_acceptance` is an internal function I'm not too fuzzed about the variable naming IF you improve the docstring. However, improving the argument names would be a great benefit.. we're not in the 70's anymore, where economy of names was a concern.

Reply:
Haha I get your points, just didn't think about it during the initial setup since all variables in the formulas are called `n` and `m` as well. `n` is very common in statistics for the sample size.